File: client/src/main/java/com/skm/sftp/client/ClientHook.java
package com.skm.sftp.client;

import com.inductiveautomation.ignition.common.script.ScriptManager;
import com.inductiveautomation.ignition.common.script.hints.PropertiesFileDocProvider;
import com.inductiveautomation.vision.api.client.AbstractClientModuleHook;

@SuppressWarnings("unused")
public class ClientHook extends AbstractClientModuleHook{

  @Override
  public void initializeScriptManager(ScriptManager manager){
    super.initializeScriptManager(manager);
    manager.addScriptModule(
        "system.sftp",
        new ClientScriptModule(),
        new PropertiesFileDocProvider()
    );
  }
}

File: client/bin/main/com/skm/sftp/client/ClientHook.kt
package com.skm.sftp.client

import com.inductiveautomation.ignition.common.script.ScriptManager
import com.inductiveautomation.ignition.common.script.hints.PropertiesFileDocProvider
import com.inductiveautomation.vision.api.client.AbstractClientModuleHook
import com.skm.sftp.AbstractScriptModule.Companion.modulePath

@Suppress("unused")
class ClientHook : AbstractClientModuleHook() {
    override fun initializeScriptManager(manager: ScriptManager) {
        super.initializeScriptManager(manager)
        manager.addScriptModule(
            modulePath,
            ClientScriptModule(),
            PropertiesFileDocProvider()
        )
    }
}